pkg/resource/deploy/step_executor.go

```diff
@@ -41,9 +41,40 @@ var (
 	errStepApplyFailed = errors.New("step application failed")
 )
 
+// The step executor operates in terms of "chains" and "antichains". A chain is set of steps that are totally ordered
+// when ordered by dependency; each step in a chain depends directly on the step that comes before it. An antichain
+// is a set of steps that is completely incomparable when ordered by dependency. The step executor is aware that chains
+// must be executed serially and antichains can be executed concurrently.
+//
+// See https://en.wikipedia.org/wiki/Antichain for more complete definitions. The below type aliases are useful for
+// documentation purposes.
+
 // A Chain is a sequence of Steps that must be executed in the given order.
 type Chain = []Step
 
+// An Antichain is a set of Steps that can be executed in parallel.
+type Antichain = []Step
+
+// A CompletionToken is a token returned by the step executor that is completed when the chain has completed execution.
+// Callers can use it to optionally wait synchronously on the completion of a chain.
+type CompletionToken struct {
```

Comment:
Out of curiosity, do any of the executor/token/etc. types need to be exported? IIRC they're now all internal to this package, right?

Reply:
I think you're right, these don't need to be exported.